lib/request.js

```diff
@@ -6,8 +6,9 @@ const path = require('path');
 const { CI_DOMAIN } = require('./ci/ci_type_parser');
 
 class Request {
-  constructor(credentials) {
+  constructor({ credentials, proxyAgent = undefined } = {}) {
```

Comment:
Why not just create a proxyAgent here (at the top level so it's shared) and use it throughout the code base?